### PR TITLE
[android] Add lint support

### LINKED ISSUE #8357: Ensure Lint behaves the same in Debug and Release
Lint behaves differently depending on the build type.
We should ensure that behaves the same in both Debug and Release in order to detect potential issues as soon as possible (e.g. #8356).

cc/ @zugaldia

### PR DESCRIPTION
**Pull request by Guardiola31337**
Fixes #8357 

- Adds lint using our current baseline as starting point

👀 @zugaldia @tobrun

### CODE REVIEW

**Comment by tobrun** [comment]
This is great! Are these rules also enforced on CI? or is that one bridge too far?

**Comment by zugaldia** [comment]
While I'm happy to merge this PR with the current baseline, by looking at [the SDK file](https://github.com/mapbox/mapbox-gl-native/blob/6cdf3ead0bd8b954681b4a8404b2f8670f6ca4d4/platform/android/MapboxGLAndroidSDK/lint-baseline.xml), it seems to me that there're many issues that are easily fixable (e.g. setting a default Locale during logging, or using `@SuppressWarnings({"MissingPermission"})` when we're sure we don't need to check for a permission).

@Guardiola31337 Could you open a follow up ticket where we'll review the current baseline to make it as small as possible? This looks like a great starter task.

**Comment by Guardiola31337** [comment]
@tobrun 
> Are these rules also enforced on CI?

Not yet, we should add a lint check step. I commented how in https://github.com/mapbox/mapbox-gl-native/issues/8357#issuecomment-309798446 👇 

> In order to accomplish this, we could add a `lint` check on CI. [Run lint with Gradle](https://developer.android.com/studio/write/lint.html#lint-task).

@zugaldia 
> While I'm happy to merge this PR with the current baseline, by looking at the SDK file, it seems to me that there're many issues that are easily fixable (e.g. setting a default Locale during logging, or using @SuppressWarnings({"MissingPermission"}) when we're sure we don't need to check for a permission).

> Could you open a follow up ticket where we'll review the current baseline to make it as small as possible? This looks like a great starter task.

Wasn't my intention, actually we could work here and in the meantime we could add lint on CI.
Cool?

**Comment by zugaldia** [comment]
>  we could work here and in the meantime we could add lint on CI.

Yeah, let's add lint to CI now with the current baseline. We can then work (separate PR?) on reducing the baseline. That way lint doesn't have to wait for us to work on the baseline.

**Comment by tobrun** [comment]
Executing lint from command line is something as:

```
./gradlew lint 
```

Though we could also expose a separate make target for it as we do for checkstyle

**Comment by Guardiola31337** [comment]
👀 ready for review @tobrun @zugaldia

File: platform/android/MapboxGLAndroidSDK/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+lint-baseline.xml
+lint/lint-baseline-local.xml
```
Comment (zugaldia) [inline platform/android/MapboxGLAndroidSDK/.gitignore]: @Guardiola31337 Why do we have different baselines for local dev and for CI? Shouldn't they both be the same?

Comment (Guardiola31337) [inline platform/android/MapboxGLAndroidSDK/.gitignore]: @zugaldia they should, but in practice baselines include absolute paths (pointing to the files containing an error) which are different in local and in CI, so we need to keep both 😞

Comment (Guardiola31337) [inline platform/android/MapboxGLAndroidSDK/.gitignore]: @zugaldia at the moment, if you run this locally and remove any error, you'll get a warning from the command line advising you to remove it from the baseline. If you removed it you should remove it too from the CI baseline (which is the only one included in the repo). Eventually, it'll be removed (when we remove all current lint errors included).

Comment (zugaldia) [inline platform/android/MapboxGLAndroidSDK/.gitignore]: @Guardiola31337 Thanks for the explanation 👍 . Could we add a big note to the XML lint files saying that? Something like "remember to update file XXXX.XML as well if you make any changes to this file".

Comment (Guardiola31337) [inline platform/android/MapboxGLAndroidSDK/.gitignore]: @zugaldia I've added the following comment
```xml
<!-- REMEMBER! First you run Lint locally you'll need to move the lint-baseline.xml file
    generated into the lint folder and called it lint-baseline-local.xml
    If you remove any error when running Lint locally, you'll get a warning from the command
    line advising you to remove it from the baseline. If you remove it (remember to remove it
    from lint-baseline-local.xml file) you should remove it too from
    lint-baseline-ci.xml (THIS FILE) which is the only one included in the repo.
    Eventually, it'll be removed (when we remove all current lint errors included). -->
```
into the different `lint-baseline-ci.xml` files.

Comment (zugaldia) [inline platform/android/MapboxGLAndroidSDK/.gitignore]: @Guardiola31337 👍

**Pull request merged**